guacamole-common-js/src/main/webapp/modules/Tunnel.js

```diff
@@ -933,6 +958,16 @@ Guacamole.WebSocketTunnel = function(tunnelURL) {
             tunnel.setState(Guacamole.Tunnel.State.UNSTABLE);
         }, tunnel.unstableThreshold);
 
+        var currentTime = new Date().getTime();
+        var pingDelay = Math.max(lastSentPing + PING_FREQUENCY - currentTime, 0);
+
+        // Ping tunnel endpoint regularly to test connection stability, sending
```

Comment:
Where is this happening regularly? I see the keepalive ping above getting triggered by receiving stuff from the server, but don't see anything like that for this. As far as I can tell it's only ever called in `socket.onopen()`. Does that happen regularly?

Reply:
Oh nevermind I see it now.